Fortinet/fortiweb/ingest/parser.yml

```diff
@@ -30,10 +32,17 @@ stages:
           network.protocol: "{{parsed_event.message.proto}}"
           rule.ruleset: "{{parsed_event.message.policy}}"
           source.geo.name: "{{parsed_event.message.srccountry}}"
-          source.ip: "{{parsed_event.message.src}}"
+          source.ip: "{{parsed_event.message.src or parsed_event.message.src_ip}}"
           source.port: "{{parsed_event.message.src_port}}"
           tls.cipher: "{{parsed_event.message.cipher_suite}}"
+          url.domain: "{{parsed_event.message.http_host}}"
           url.path: "{{parsed_event.message.http_url}}"
           url.username: "{{parsed_event.message.user_name}}"
           user.name: "{{parsed_event.message.user}}"
           user_agent.original: "{{parsed_event.message.http_agent}}"
+          user.id: "{{parsed_event.message.user_id}}"
+
+      - set:
+          user.name: "{{parsed_event.message.user_name.split('@')[0]}}"
+          user.domain: "{{parsed_event.message.user_name.split('@')[1]}}"
+        filter: "{{parsed_event.message.user_name != None and '@' in parsed_event.message.user_name}}"
```

Comment:
Can you add the case that `user_name` contains only the name of the user (and not the email of the user)?
In this case `user_name` go into `user.name`